server/src/main/java/org/jboss/as/server/deployment/DeploymentHandlerUtils.java

```diff
@@ -47,6 +48,8 @@
  */
 public abstract class DeploymentHandlerUtils {
 
+    public static final Pattern ARCHIVE_PATTERN = Pattern.compile("^.*\\.(?:(?:[SsWwJjEeRr][Aa][Rr])|(?:[Ww][Aa][Bb])|(?:[Ee][Ss][Aa]))$");
```

Comment:
Out of curiosity, could you describe the pattern?
It seems to matches more than ear, jar, war, sar, rar. What are the last 2 groups for?